scipy/signal/_splinemodule.h

```diff
@@ -1,25 +1,340 @@
+#pragma once
 #include "Python.h"
-#include "numpy/arrayobject.h"
-#include <math.h>
 
+#ifndef M_PI
+#define M_PI           3.14159265358979323846  /* pi */
+#endif
 
-#define PYERR(message) do {PyErr_SetString(PyExc_ValueError, message); goto fail;} while(0)
+#ifndef PyArray_MIN
+#define PyArray_MIN(a,b) (((a)<(b))?(a):(b))
+#endif
 
-static void convert_strides(npy_intp*,npy_intp*,int,int);
+#define PYERR(message)                                                                                                 \
+    do {                                                                                                               \
+        PyErr_SetString(PyExc_ValueError, message);                                                                    \
+        goto fail;                                                                                                     \
+    } while (0)
 
-extern int S_SYM_IIR1_initial(float, float*, float*, int, int, float);
-extern int S_SYM_IIR2_initial_fwd(double, double, float*, float*, int, int, float);
-extern int S_SYM_IIR2_initial_bwd(double, double, float*, float*, int, int, float);
-extern int S_separable_2Dconvolve_mirror(float*,float*,int,int,float*,float*,int,int,npy_intp*,npy_intp*);
 
-extern int D_SYM_IIR1_initial(double, double*, double*, int, int, double);
-extern int D_SYM_IIR2_initial_fwd(double, double, double*, double*, int, int, double);
-extern int D_SYM_IIR2_initial_bwd(double, double, double*, double*, int, int, double);
-extern int D_separable_2Dconvolve_mirror(double*,double*,int,int,double*,double*,int,int,npy_intp*,npy_intp*);
+template <typename T, typename C>
+int _sym_iir1_initial(C z1, C *x, C *yp0, int M, int N, T precision) {
```

Comment:
Could we have a basic docstring here? What are the inputs, what does the function do?

Reply:
This is still on my wishlist